src/rgb/contract/amount.rs

```diff
@@ -503,15 +503,149 @@ mod test {
         0xbf, 0x1d, 0x10, 0x75, 0xc0, 0xf6, 0x9c,
     ];
 
+    static AMOUNT_64: [u8; 43] = [
+        0x3, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x20, 0x0, 0xab, 0xe8, 0x9d, 0x73, 0xbd,
+        0x1c, 0x25, 0x6d, 0x3c, 0x94, 0x94, 0xda, 0x5, 0xcc, 0x29, 0x7e, 0x34, 0xc3, 0xed, 0xfb,
+        0x6, 0xdb, 0x6f, 0xe4, 0xdf, 0x6f, 0x28, 0x6e, 0x5d, 0xf6, 0xce, 0xfe,
+    ];
+
+    static COMMIT_SUM: [u8; 35] = [
+        0x21, 0x0, 0x9, 0x36, 0x2d, 0xe0, 0xe6, 0x5f, 0x65, 0x31, 0xe9, 0x60, 0x5, 0xcc, 0xf7,
+        0x15, 0x2c, 0x7d, 0xa9, 0x16, 0x8a, 0x2f, 0x32, 0x25, 0x52, 0xa2, 0x9b, 0xe0, 0xb5, 0xc,
+        0x19, 0xc2, 0x4d, 0x98, 0x95,
+    ];
+
     #[test]
     fn test_amount() {
-        test_encode!(AMOUNT);
+        // Test encoding decoding
+        test_encode!((AMOUNT, Revealed));
+        test_encode!((CONFIDENTIAL_AMOUNT, Confidential));
+
+        // Test commitment
         assert!(test_confidential::<Revealed>(&AMOUNT, &CONFIDENTIAL_AMOUNT).is_ok());
+
+        // Test comparison
+        let revealed_64 = Revealed::strict_decode(&AMOUNT_64[..]).unwrap();
+        let old_revealed = Revealed::strict_decode(&AMOUNT[..]).unwrap();
+        assert_eq!(revealed_64.cmp(&old_revealed), Ordering::Less);
+        assert_eq!(
+            revealed_64.partial_cmp(&old_revealed).unwrap(),
+            Ordering::Less
+        );
+        let coded_conf = Confidential::strict_decode(&CONFIDENTIAL_AMOUNT[..]).unwrap();
+        let old_conf = old_revealed.conceal();
+        let new_conf = revealed_64.conceal();
+        assert_eq!(coded_conf, old_conf);
+        assert_ne!(old_conf, new_conf);
+        assert_eq!(old_conf.cmp(&new_conf), Ordering::Greater);
+        assert_eq!(old_conf.partial_cmp(&new_conf).unwrap(), Ordering::Greater);
+
+        // Test confidential addition
+        assert!(coded_conf.verify_bullet_proof().is_ok());
+        let new_commit = new_conf.commitment;
+        let sum = old_conf.add(new_commit);
+        let commit_sum =
+            secp256k1zkp::pedersen::Commitment::strict_decode(&COMMIT_SUM[..]).unwrap();
+        assert_eq!(sum, commit_sum);
+    }
+
+    #[test]
+    fn test_commit_sum() {
+        let postive = [1u64, 2u64, 3u64, 4u64, 5u64];
+        let negetive = [7u64, 8u64];
+
+        let postive_commits = postive
+            .iter()
+            .map(|amount| {
+                Revealed::with_amount(*amount as Amount, &mut rand::thread_rng())
+                    .conceal()
+                    .commitment
+            })
+            .collect::<Vec<secp256k1zkp::pedersen::Commitment>>();
+
+        let negetive_commits = negetive
+            .iter()
+            .map(|amount| {
+                Revealed::with_amount(*amount as Amount, &mut rand::thread_rng())
+                    .conceal()
+                    .commitment
+            })
+            .collect::<Vec<secp256k1zkp::pedersen::Commitment>>();
+
+        //TODO: Investigate why commit sum verification fails
```

Comment:
It fails since you choose random blinding factors, while they also must add to 0 — see how it's done in https://github.com/LNP-BP/rust-lnpbp/blob/master/src/rgb/contract/assignments.rs#L49 method (and it is the only reason we need that method)

Reply:
Hmm, I am still not sure how to do this. Can't seem to find any usage of `verify_commit_sum()` in the above reference.

Reply:
Just leave this test as it is; the only thing you should do is to zero-balance blinding factors with secp256k1 function like it is done in the Assignments function from the link above

Reply:
So, remove the last blinding factor and replace it with the result of the following code:
```rust
       let blinding_correction = SECP256K1_ZKP
            .blind_sum(blinding_inputs.clone(), blinding_factors.clone())
            .expect("SECP256K1_ZKP failure has negligible probability");
        blinding_factors.push(blinding_correction);
```

Reply:
I am trying this. Lets see.